src/styles/calendar.scss

```diff
@@ -0,0 +1,50 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  width: $cellSize * 7 + $padding * 2 + $gap * 6;
+  gap: $gap;
+  padding: $padding;
+
+  &__day {
+    display: flex;
+    background-color: #eee;
+    border: 1px solid black;
+    width: $cellSize;
+    height: $cellSize;
+    justify-content: center;
+    align-items: center;
+    transition: 0.5s;
+  }
+
+  &__day::before {
+    font-family: Arial;
+    font-size: 30px;
+  }
+
+  @for $i from 1 through 31 {
+    &__day:nth-child(#{$i})::before {
+      content: '#{$i}';
+    }
+  }
+
+  @for $i from 28 through 31 {
+    &--month-length--#{$i} &__day:nth-child(n + #{$i + 1}) {
+      display: none;
+    }
+  }
+
+  &__day:hover {
+    cursor: pointer;
+    background-color: pink;
+    transform: translateY(-20px);
+  }
+}
+
+$days: ('mon': 0, 'tue': 1, 'wed': 2, 'thu': 3, 'fri': 4, 'sat': 5, 'sun': 6);
```

Comment:
It would be good to keep all the variables together, it will increase the readability and maintainability of your code.